src/type-checker.cc

```diff
@@ -344,15 +349,11 @@ Result TypeChecker::CheckOpcode3(Opcode opcode,
                                  const Limits* limits1,
                                  const Limits* limits2,
                                  const Limits* limits3) {
-  bool has_address_operands = limits1 || limits2 || limits3;
-  Result result =
-      PopAndCheck3Types(opcode.GetMemoryParam(opcode.GetParamType1(), limits1,
-                                              has_address_operands),
-                        opcode.GetMemoryParam(opcode.GetParamType2(), limits2,
-                                              has_address_operands),
-                        opcode.GetMemoryParam(opcode.GetParamType3(), limits3,
-                                              has_address_operands),
-                        opcode.GetName());
+  Result result = PopAndCheck3Types(
+      GetMemoryParam(opcode.GetParamType1(), limits1, limits1 != nullptr),
+      GetMemoryParam(opcode.GetParamType2(), limits2, limits2 != nullptr),
+      GetMemoryParam(opcode.GetParamType3(), limits3, limits3 != nullptr),
```

Comment:
This feels a bit weird for now (if each time the third argument is really just `secondarg != nullptr`, we could omit the third parameter to GetMemoryParam completely). But I guess it will need to make the decision in a more sophisticated way once this correctly implements memory.copy across different-type memories.

Reply:
Hmm.. you are right.. I simplified some more.